### PR TITLE
Convert to auto-plugin (closes #14)

### LINKED ISSUE #14: Make sbt-jmh an AutoPlugin
Make it an autoplugin :-) http://www.scala-sbt.org/release/docs/Plugins.html

### PR DESCRIPTION
**Pull request by hseeberger**
Changes made for auto-plugin. I also added the sbt-jhm-tester subdirectory: just cd into it, open sbt and test sbt-jhm, e.g. by executing `run`.

Additionally I (hopefully) improved/simplified some things.

### CODE REVIEW

**Comment by ktoso** [comment]
Thanks!

File: src/main/scala/pl/project13/scala/sbt/JmhPlugin.scala [hunk 1]
```diff
@@ -2,18 +2,37 @@ package pl.project13.scala.sbt
...
+    val outputTarget = settingKey[File]("Directory where the bytecode to be consumed and generated sources should be written to (`target` or sometimes `target/scala-2.10`)")
+
+    val generatorType = settingKey[String]("Benchmark code generator type. Available: `default`, `reflection` or `asm`.")
+
+    val generateInstrumentedClasses = taskKey[Seq[File]]("Generate instrumented JMH code")
+  }
 
   import JmhKeys._
 
-  lazy val jmhSettings = Seq(
```
Comment (densh) [inline src/main/scala/pl/project13/scala/sbt/JmhPlugin.scala]: This change invalidates README and activator project template.

Comment (densh) [inline src/main/scala/pl/project13/scala/sbt/JmhPlugin.scala]: I'd say it warrants bump of the first version (e.g. `0.2.0`) as API is slightly different now and requires changes to existing builds.

Comment (ktoso) [inline src/main/scala/pl/project13/scala/sbt/JmhPlugin.scala]: Sounds like a plan, I can roll the fixes tonight https://github.com/ktoso/sbt-jmh/issues/38
Unless someone beats me to it :)

Comment (densh) [inline src/main/scala/pl/project13/scala/sbt/JmhPlugin.scala]: Cool, thanks!

**Pull request merged**